From 449e1ebabffe79c94df1836888a3954b01820997 Mon Sep 17 00:00:00 2001
From: Adrian Schmutzler <freifunk@adrianschmutzler.de>
Date: Sat, 26 Jan 2019 13:57:33 +0100
Subject: [PATCH] ar71xx: Fix 5 GHz MAC address for Archer C60 v2

Looks like C60 v2 needs this the MAC address to be calculated
manually, while the C60 v1 gets it correctly without manual
interference.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>
---
 .../ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata     | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata b/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
index 53f53775fd..d21d4134d4 100644
--- a/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
+++ b/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
@@ -168,13 +168,18 @@ case "$FIRMWARE" in
 	archer-c58-v1|\
 	archer-c59-v1|\
 	archer-c60-v1|\
-	archer-c60-v2|\
 	cf-e355ac-v2|\
 	cf-e375ac)
 		ath10kcal_extract "art" 20480 12064
 		ln -sf /lib/firmware/ath10k/pre-cal-pci-0000\:00\:00.0.bin \
 			/lib/firmware/ath10k/QCA9888/hw2.0/board.bin
 		;;
+	archer-c60-v2)
+		ath10kcal_extract "art" 20480 12064
+		ln -sf /lib/firmware/ath10k/pre-cal-pci-0000\:00\:00.0.bin \
+			/lib/firmware/ath10k/QCA9888/hw2.0/board.bin
+		ath10kcal_patch_mac $(macaddr_add $(cat /sys/class/net/eth0/address) -1)
+		;;
 	cf-e385ac)
 		ath10kcal_extract "art" 20480 12064
 		;;
-- 
2.11.0

